File: sales-savvy-frontend/src/services/adminProductService.js
import axios from "axios";

const API = "http://localhost:8080/api/admin/products";

export const getAllProducts = () =>
    axios.get(API, {
        withCredentials: true,
    });

export const getProductById = (id) =>
    axios.get(`${API}/${id}`, {
        withCredentials: true,
    });

export const addProduct = (formData) =>
    axios.post(API, formData, {
        withCredentials: true,
    });

export const updateProduct = (id, formData) =>
    axios.put(`${API}/${id}`, formData, {
        withCredentials: true,
    });

export const deleteProduct = (id) =>
    axios.delete(`${API}/${id}`, {
        withCredentials: true,
    });

export const deleteProductImage = (productId, imageId) =>
    axios.delete(
        `${API}/${productId}/images/${imageId}`,
        {
            withCredentials: true,
        }
    );

File: SalesSavvy-Backend/src/main/java/com/salessavvy/app/services/AdminProductService.java
package com.salessavvy.app.services;

import java.io.IOException;
import java.util.List;

import com.salessavvy.app.dto.request.ProductRequestDTO;
import com.salessavvy.app.dto.response.ProductResponseDTO;

public interface AdminProductService {

    ProductResponseDTO addProduct(
            ProductRequestDTO request)
            throws IOException;

    ProductResponseDTO updateProduct(
            Integer productId,
            ProductRequestDTO request)
            throws IOException;

    void deleteProduct(
            Integer productId)
            throws IOException;

    void deleteProductImage(
            Integer productId,
            Integer imageId)
            throws IOException;

    List<ProductResponseDTO> getAllProducts();

    ProductResponseDTO getProductById(
            Integer productId);
}